### PR TITLE
Minor: SMJ fuzz tests fix for rowcounts

### LINKED ISSUE #10886: SMJ producing different results than HashJoin when doing a semi join
### Describe the bug

As a part of #10728 we have added tests for semi join, and on the first merge on main they failed (they always passed). I have ignored that tests for now but that failure suggests a bug . It only happened once in several builds, and on the AMD64 architecture

```
---- fuzz_cases::join_fuzz::test_semi_join_1k_filtered stdout ----
thread 'fuzz_cases::join_fuzz::test_semi_join_1k_filtered' panicked at datafusion/core/tests/fuzz_cases/join_fuzz.rs:416:13:
assertion `left == right` failed: SortMergeJoinExec and HashJoinExec produced different row counts
  left: 100
 right: 99
stack backtrace:
   0: rust_begin_unwind
```

see https://github.com/apache/datafusion/actions/runs/9475682375/job/26107451352

### To Reproduce

https://github.com/apache/datafusion/actions/runs/9475682375/job/26107451352

### Expected behavior

Should pass

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by comphead**
## Which issue does this PR close?

Closes #10886.

## Rationale for this change

The fuzz tests for filtered SMJ introduced in https://github.com/apache/datafusion/pull/10728 has a flaw when calculating the rowcounts, it fact it caculcated batches which are not always the same and led to false negatives.

### CODE REVIEW

**Comment by comphead** [comment]
@edmondop cc

**Comment by edmondop** [comment]
Thanks, why does changing the threshold fix the problem?

**Comment by comphead** [comment]
> Thanks, why does changing the threshold fix the problem?

it doesn't fix the threshold, the method name renamed, I think it takes less than 100 right?

**Review by edmondop — APPROVED** [review]

File: datafusion/core/tests/fuzz_cases/join_fuzz.rs [hunk 1]
```diff
@@ -414,21 +422,20 @@ impl JoinFuzzTestCase {
...
-                .enumerate()
-            {
-                assert_eq!(
-                    (i, smj_line),
-                    (i, hj_line),
-                    "SortMergeJoinExec and HashJoinExec produced different results"
-                );
+            // row level compare if any of joins returns the result
+            // the reason is different formatting when there is no rows
+            if smj_rows > 0 || hj_rows > 0 {
```
Comment (alamb) [inline datafusion/core/tests/fuzz_cases/join_fuzz.rs:427]: I double checked that the assert on row count is done higher up 👍

**Review by alamb — APPROVED** [review]
Nice work @comphead  -- thank you for the review @edmondop

**Pull request merged**